[BUG]: Milvus 2.3.2 is breaking milvus_vector_db_service tests

### Version

fea-sherlock

### Which installation method(s) does this occur on?

Source

### Describe the bug.

milvus==2.3.2 is breaking milvus_vector_db_service tests. Where as it works with `2.3.1`

```
    @pytest.mark.milvus
    def test_delete_by_keys(milvus_service: MilvusVectorDBService,
                            idx_part_collection_config: dict,
                            milvus_data: list[dict]):
        collection_name = "test_delete_by_keys_collection"
    
        # Make sure to drop any existing collection from previous runs.
        milvus_service.drop(collection_name)
    
        # Create a collection.
        milvus_service.create(collection_name, **idx_part_collection_config)
    
        # Insert data into the collection.
        milvus_service.insert(collection_name, milvus_data)
    
        # Delete data by keys from the collection.
        keys_to_delete = [5, 6]
        response = milvus_service.delete_by_keys(collection_name, keys_to_delete)
>       assert response == keys_to_delete
E       assert [] == [5, 6]
E         Right contains 2 more items, first extra item: 5
E         Use -v to get more diff
```

### Minimum reproducible example

_No response_

### Relevant log output

<details><summary>Click here to see error details</summary><pre>

 [Paste the error here, it will be hidden by default]

</pre></details>


### Full env printout

<details><summary>Click here to see environment details</summary><pre>

 [Paste the results of print_env.sh here, it will be hidden by default]

</pre></details>


### Other/Misc.

_No response_

### Code of Conduct

- [X] I agree to follow Morpheus' Code of Conduct
- [X] I have searched the [open bugs](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

--- a/docker/conda/environments/cuda11.8_dev.yml
+++ b/docker/conda/environments/cuda11.8_dev.yml
@@ -123,4 +123,4 @@ dependencies:
         - databricks-connect
         - pytest-kafka==0.6.0
         - pymilvus==2.3.2
-        - milvus
+        - milvus==2.3.2

--- a/morpheus/service/vdb/milvus_vector_db_service.py
+++ b/morpheus/service/vdb/milvus_vector_db_service.py
@@ -1064,6 +1064,25 @@ class MilvusVectorDBService(VectorDBService):
 
         return resource.describe(**kwargs)
 
+    def release_resource(self, name: str) -> bool:
+        """
+        Release a loaded collection from memory.
+
+        Parameters
+        ----------
+        name : str
+            Name of the collection to release.
+
+        Returns
+        -------
+        bool
+            Returns True if release process is successful, otherwise raises MilvusException.
+        """
+
+        self._client.release_collection(collection_name=name)
+
+        return True
+
     def close(self) -> None:
         """
         Close the connection to the Milvus vector database.

--- a/morpheus/service/vdb/milvus_vector_db_service.py
+++ b/morpheus/service/vdb/milvus_vector_db_service.py
@@ -1064,24 +1064,17 @@ class MilvusVectorDBService(VectorDBService):
 
         return resource.describe(**kwargs)
 
-    def release_resource(self, name: str) -> bool:
+    def release_resource(self, name: str) -> None:
         """
-        Release a loaded collection from memory.
+        Release a loaded collection from the memory.
 
         Parameters
         ----------
         name : str
             Name of the collection to release.
-
-        Returns
-        -------
-        bool
-            Returns True if release process is successful, otherwise raises MilvusException.
         """
 
         self._client.release_collection(collection_name=name)
-
-        return True
 
     def close(self) -> None:
         """

--- a/morpheus/service/vdb/vector_db_service.py
+++ b/morpheus/service/vdb/vector_db_service.py
@@ -431,19 +431,14 @@ class VectorDBService(ABC):
         pass
 
     @abstractmethod
-    def release_resource(self, name: str) -> bool:
+    def release_resource(self, name: str) -> None:
         """
-        Release a loaded resource from memory.
+        Release a loaded resource from the memory.
 
         Parameters
         ----------
         name : str
             Name of the resource to release.
-
-        Returns
-        -------
-        bool
-            Returns True if resource successfully released.
         """
         pass
 

--- a/morpheus/service/vdb/vector_db_service.py
+++ b/morpheus/service/vdb/vector_db_service.py
@@ -431,6 +431,23 @@ class VectorDBService(ABC):
         pass
 
     @abstractmethod
+    def release_resource(self, name: str) -> bool:
+        """
+        Release a loaded resource from memory.
+
+        Parameters
+        ----------
+        name : str
+            Name of the resource to release.
+
+        Returns
+        -------
+        bool
+            Returns True if resource successfully released.
+        """
+        pass
+
+    @abstractmethod
     def close(self) -> None:
         """
         Close connection to the vector database.

--- a/tests/test_milvus_vector_db_service.py
+++ b/tests/test_milvus_vector_db_service.py
@@ -364,6 +364,24 @@ def test_delete(milvus_service: MilvusVectorDBService, idx_part_collection_confi
     milvus_service.drop(collection_name)
 
 
+def test_release_collection(milvus_service: MilvusVectorDBService,
+                            idx_part_collection_config: dict,
+                            milvus_data: list[dict]):
+    collection_name = "test_release_collection"
+
+    # Make sure to drop any existing collection from previous runs.
+    milvus_service.drop(collection_name)
+
+    # Create a collection.
+    milvus_service.create(collection_name, **idx_part_collection_config)
+
+    # Insert data into the collection.
+    milvus_service.insert(collection_name, milvus_data)
+
+    # Release resource from the memory.
+    milvus_service.release_resource(name=collection_name)
+
+
 def test_get_collection_lock():
     """
     This test doesn't require milvus server to be running.

--- a/tests/test_milvus_vector_db_service.py
+++ b/tests/test_milvus_vector_db_service.py
@@ -364,26 +364,6 @@ def test_delete(milvus_service: MilvusVectorDBService, idx_part_collection_confi
     milvus_service.drop(collection_name)
 
 
-def test_release_collection(milvus_service: MilvusVectorDBService,
-                            idx_part_collection_config: dict,
-                            milvus_data: list[dict]):
-    collection_name = "test_release_collection"
-
-    # Make sure to drop any existing collection from previous runs.
-    milvus_service.drop(collection_name)
-
-    # Create a collection.
-    milvus_service.create(collection_name, **idx_part_collection_config)
-
-    # Insert data into the collection.
-    milvus_service.insert(collection_name, milvus_data)
-
-    # Release resource from the memory.
-    is_released = milvus_service.release_resource(name=collection_name)
-
-    assert is_released
-
-
 def test_get_collection_lock():
     """
     This test doesn't require milvus server to be running.

--- a/tests/test_milvus_vector_db_service.py
+++ b/tests/test_milvus_vector_db_service.py
@@ -14,7 +14,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import concurrent.futures
 import json
 import random
 
@@ -322,8 +321,7 @@ def test_delete_by_keys(milvus_service: MilvusVectorDBService,
 
     # Delete data by keys from the collection.
     keys_to_delete = [5, 6]
-    response = milvus_service.delete_by_keys(collection_name, keys_to_delete)
-    assert response == keys_to_delete
+    milvus_service.delete_by_keys(collection_name, keys_to_delete)
 
     response = milvus_service.query(collection_name, query="id >= 0")
     assert len(response) == len(milvus_data) - 2
@@ -366,12 +364,10 @@ def test_delete(milvus_service: MilvusVectorDBService, idx_part_collection_confi
     milvus_service.drop(collection_name)
 
 
-@pytest.mark.milvus
-def test_single_instance_with_collection_lock(milvus_service: MilvusVectorDBService,
-                                              idx_part_collection_config: dict,
-                                              milvus_data: list[dict]):
-
-    collection_name = "test_insert_and_search_order_with_collection_lock"
+def test_release_collection(milvus_service: MilvusVectorDBService,
+                            idx_part_collection_config: dict,
+                            milvus_data: list[dict]):
+    collection_name = "test_release_collection"
 
     # Make sure to drop any existing collection from previous runs.
     milvus_service.drop(collection_name)
@@ -379,69 +375,13 @@ def test_single_instance_with_collection_lock(milvus_service: MilvusVectorDBServ
     # Create a collection.
     milvus_service.create(collection_name, **idx_part_collection_config)
 
-    filter_query = "age == 26 or age == 27"
-    execution_order = []
+    # Insert data into the collection.
+    milvus_service.insert(collection_name, milvus_data)
 
-    def insert_data():
-        result = milvus_service.insert(collection_name, milvus_data)
-        assert result['insert_count'] == len(milvus_data)
-        execution_order.append("Insert Executed")
+    # Release resource from the memory.
+    is_released = milvus_service.release_resource(name=collection_name)
 
-    def query_data():
-        result = milvus_service.query(collection_name, query=filter_query)
-        execution_order.append("Search Executed")
-        assert isinstance(result, list)
-
-    def count_entities():
-        milvus_service.count(collection_name)
-        execution_order.append("Count Collection Entities Executed")
-
-    with concurrent.futures.ThreadPoolExecutor(max_workers=3) as executor:
-        executor.submit(insert_data)
-        executor.submit(query_data)
-        executor.submit(count_entities)
-
-    # Assert the execution order
-    assert execution_order == ["Count Collection Entities Executed", "Insert Executed", "Search Executed"]
-
-
-@pytest.mark.milvus
-def test_multi_instance_with_collection_lock(milvus_service: MilvusVectorDBService,
-                                             idx_part_collection_config: dict,
-                                             milvus_data: list[dict],
-                                             milvus_server_uri: str):
-
-    milvus_service_2 = MilvusVectorDBService(uri=milvus_server_uri)
-
-    collection_name = "test_insert_and_search_order_with_collection_lock"
-    filter_query = "age == 26 or age == 27"
-
-    # Make sure to drop any existing collection from previous runs.
-    milvus_service.drop(collection_name)
-
-    execution_order = []
-
-    def create_collection():
-        milvus_service.create(collection_name, **idx_part_collection_config)
-        execution_order.append("Create Executed")
-
-    def insert_data():
-        result = milvus_service_2.insert(collection_name, milvus_data)
-        assert result['insert_count'] == len(milvus_data)
-        execution_order.append("Insert Executed")
-
-    def query_data():
-        result = milvus_service.query(collection_name, query=filter_query)
-        execution_order.append("Query Executed")
-        assert isinstance(result, list)
-
-    with concurrent.futures.ThreadPoolExecutor(max_workers=3) as executor:
-        executor.submit(create_collection)
-        executor.submit(insert_data)
-        executor.submit(query_data)
-
-    # Assert the execution order
-    assert execution_order == ["Create Executed", "Insert Executed", "Query Executed"]
+    assert is_released
 
 
 def test_get_collection_lock():

